client/db/src/lib.rs

```diff
@@ -672,8 +681,16 @@ impl<Block: BlockT> sc_state_db::NodeDb for StorageDb<Block> {
 	type Error = io::Error;
 	type Key = [u8];
 
-	fn get(&self, key: &[u8]) -> Result<Option<Vec<u8>>, Self::Error> {
-		self.db.get(columns::STATE, key).map(|r| r.map(|v| v.to_vec()))
+	fn get(&self, child_info: &ChildInfo, key: &[u8]) -> Result<Option<Vec<u8>>, Self::Error> {
+		if child_info.is_top_trie() {
```

Comment:
Okay, this is getting weird IMHO.

We pass a `ChildInfo` that says if we should access the top trie?

Some out of the context question, is top trie the wording we use? I still find it confusing to speak from top/child trie.

Reply:
Can we not pass some `enum` as `info`?

Reply:
About top trie wording, this is rather used in state-machine, but it may not be the best wording, maybe `root_trie` will be better?

ChildInfo is an enum, I just did not want to create a special one for the top trie, the reason for that is that technically the top trie is the child trie without a unique id, it does not have a specific behavior regarding the storage. So this check is a small optimization case, running it with the other condition works.

Giving it more thought, top trie mainly differs by the fact that it does not attach to a parent trie: in state-machine I do some redirection that repercute child calls to parent if it is top_trie, those are bad as it ignores the fact that we give a storagekey where the root will not be attached.
I will switch them to fail unless the parent key is empty.

Reply:
Yeah, I know that the difference is not that big. But I still think that having everywhere the distinction and than we use `ChildInfo`. Maybe we could just rename this to `TrieInfo`?

Regarding `root_trie`, IMHO that sounds better.

Reply:
Yes `Child` is not very happy naming, `Trie` is not either because it can potentially be a binary tree or any kind of structure that maintain a state with data.
So I could propose a renaming to 'StateInfo', and replace 'root_trie' by 'root_state'.
I really like it, but it propagates to the whole code base rather quickly (removing every 'child' to make 'child' function the default case function and default case function the 'root' specific function), that can be a pretty big move in term of documentation.